redis/conn.go

```diff
@@ -143,6 +163,55 @@ func Dial(network, address string, options ...DialOption) (Conn, error) {
 	return c, nil
 }
 
+var pathDBRegexp = regexp.MustCompile(`/(\d)\z`)
+
+// DialURL connects to a Redis server at the given URL using the Redis
+// URI scheme. URLs should follow the draft IANA specification for the
+// scheme (https://www.iana.org/assignments/uri-schemes/prov/redis).
+func DialURL(u *url.URL, options ...DialOption) (Conn, error) {
+	if u.Scheme != "redis" {
+		return nil, fmt.Errorf("invalid redis URL scheme: %s", u.Scheme)
+	}
+
+	// As per the IANA draft spec, the host defaults to localhost and
+	// the port defaults to 6379
+	host, port, err := net.SplitHostPort(u.Host)
+	if err != nil {
```

Comment:
Parsing error text is fragile. Assume that the port is not specified if err != nil.  

    if err != nil {
        host = u.Host
        port = "6379"
    } else if host == "" {
        host = "localhost"
    }